### ISSUE TITLE
[Request] implement csv support with "column" from core-utils

### ISSUE THREAD

**Issue by m040601** [wofr06/lesspipe.sh#136]
TLDR: "column" from the 30 year old "util-linux" package is "enough" for .csv files.

```
column -t -s "," my-well-behaved-csv-file.csv
```

or

```
cat my-well-behaved-csv-file.csv | column -t -s "," 
```
Would be nice if lesspipe also used it.

--- 

Currently lesspipe relys on either "csvlook" (from the python package csvkit) or pandoc

```
                csv)
                        { has_cmd csvlook && cmd=(csvlook "$1"); } ||
                        { has_cmd pandoc && cmd=(pandoc -f csv -t plain "$1"); } ;;
```

In the case that neither is present, lesspipe just splashes a little bit of color, painting  each "column" in a different color.

Now, both "csvkit" and "pandoc" are nice and powerfull tools. But they are "big tools". Install is 100 Mega. They are also not 100% available tools. Not everybody has them installed.

Now I just remembered about the utility "column" from the "util-linux" package.
This is something that 99.9% of user should already have their systems. It's one of those, small, powerfull, do one thing and do it well little things. That we sometimes forget about.

The "classic" way most people remember to use it is in a pipe, for tabulating input from stdin. Somethig like

```
mount | column -t
```

```
       -t, --table
           Determine the number of columns the input contains and create a
           table. Columns are delimited with whitespace, by default, or
           with the characters supplied using the --output-separator
           option. Table output is useful for pretty-printing.
```

For well behaved standard .csv files, if you  use the "-s" option,

```
       -s, --separator separators
           Specify the possible input item delimiters (default is
           whitespace).
```

You get a very nice table display,

```
cat  my-maybe-well-behaved-csv-file.csv
```

```
FRUIT,COLOR,PRICE
bananas,yellow,cheap
apples,green,"medium cheap"
mangos,red,"very expensive
exotic asian pears,blue,unknow price
```

```
column -t -s "," my-mabye-well-behaved-csv-file.csv
```

```
FRUIT               COLOR   PRICE
bananas             yellow  cheap
apples              green   "medium cheap"
mangos              red     "very expensive
exotic asian pears  blue    unknow price
```

As you can see it even handled my inconsistencies with multiword cells and quotes.
It will do the same for ".tsv" files if you can trust that those are "really" tab separated files.

This 30 year old tool can even do much more. And  in recent years even got json support and more. Worth checking out the man pages

https://man.archlinux.org/man/column.1
https://en.wikipedia.org/wiki/Util-linux

Apropos:

Other usefull powerfull "classics" that people also forget are "pr", "tr", "paste"  "fold" etc from coreutils. Very usefull for reshuffling and massaging your plain text displays in pipes. https://en.wikipedia.org/wiki/List_of_GNU_Core_Utilities_commands

Related past issues:

https://github.com/wofr06/lesspipe/issues/126
https://github.com/wofr06/lesspipe/issues/80

**Comment by wofr06** [comment]
I made now the csv display using `column -o "|"  -s",;<TAB>" -t` the default and removed cvslook. I left pandoc in, although it probably never gets called, as column is in the core utils and hence always installed. This patch improves #80 and #126. As discussed there wrapping long lines can be suppressed by typing `-S ENTER ` inside of the running less program. Thanks a lot for your suggestion.
